src/phoenix/server/api/routers/v1/datasets.py

```diff
@@ -25,19 +25,181 @@
+
+
+async def list_datasets(request: Request) -> Response:
+    """
+    summary: List datasets with cursor-based pagination
+    operationId: listDatasets
+    tags:
+      - datasets
+    parameters:
+      - in: query
+        name: cursor
+        required: false
+        schema:
+          type: string
+        description: Cursor for pagination
+      - in: query
+        name: limit
+        required: false
+        schema:
+          type: integer
+          default: 10
+    responses:
+      200:
+        description: A paginated list of datasets
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                next_cursor:
+                  type: string
+                data:
+                  type: array
+                  items:
+                    type: object
+                    properties:
+                      id:
+                        type: string
+                      name:
+                        type: string
+                      description:
+                        type: string
+                      metadata:
+                        type: object
+                      created_at:
+                        type: string
+                        format: date-time
+                      updated_at:
+                        type: string
+                        format: date-time
+      404:
+        description: No datasets found
+    """
+    cursor = request.query_params.get("cursor")
+    limit = int(request.query_params.get("limit", 10))
+    async with request.app.state.db() as session:
+        query = select(models.Dataset).order_by(models.Dataset.id.desc())
+
+        if cursor:
+            try:
+                cursor_id = GlobalID.from_id(cursor).node_id
+                query = query.filter(models.Dataset.id <= int(cursor_id))
+            except ValueError:
+                return Response(
+                    content=f"Invalid cursor format: {cursor}",
+                    status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+                )
+
+        query = query.limit(limit + 1)
+        result = await session.execute(query)
+        datasets = result.scalars().all()
+
+        if not datasets:
+            return JSONResponse(content={"next_cursor": None, "data": []}, status_code=200)
+
+        next_cursor = None
+        if len(datasets) == limit + 1:
+            next_cursor = str(GlobalID(NODE_NAME, str(datasets[-1].id)))
+            datasets = datasets[:-1]
+
+        data = []
+        for dataset in datasets:
+            data.append(
+                {
+                    "id": str(GlobalID(NODE_NAME, str(dataset.id))),
+                    "name": dataset.name,
+                    "description": dataset.description,
+                    "metadata": dataset.metadata_,
+                    "created_at": dataset.created_at.isoformat(),
+                    "updated_at": dataset.updated_at.isoformat(),
+                }
```

Comment:
How come we are not returning example count here? Wouldn't the user be expecting it?

Reply:
my thought is checking for example count is too expensive for the list method, I'm not sure though

Reply:
I think we should be consistent with our delivery to the users. User abrasion is probably a more expensive operation.

Reply:
ok, sounds good

Reply:
discussed with @mikeldking offline, we're going to hold off on displaying counts until requested